nixos/modules/services/databases/neo4j.nix

```diff
@@ -36,81 +89,414 @@ let
+      type = with types; attrsOf (submodule {
+        options = {
+
+          allowKeyGeneration = mkOption {
+            type = types.bool;
+            default = false;
+            description = ''
+              Allows the generation of a private key and associated self-signed
+              certificate. Only performed when both objects cannot be found for
+              this policy. It is recommended to turn this off again after keys
+              have been generated.
+
+              The public certificate is required to be duplicated to the
+              directory holding trusted certificates as defined by the
+              <literal>trustedDir</literal> option.
+
+              Keys should in general be generated and distributed offline by a
+              trusted certificate authority and not by utilizing this mode.
+            '';
+          };
+
+          baseDirectory = mkOption {
+            type = types.path;
+            default = "${cfg.directories.certificates}/__policyname__";
+            description = ''
+              The mandatory base directory for cryptographic objects of this
+              policy, where "__policyname__" in the default path is replaced by
+              the attribute name of this policy.
+
+              It is also possible to override each individual
+              configuration with absolute paths. See the
+              <literal>privateKey</literal> and <literal>publicCertificate</literal>
+              policy options.
+            '';
+          };
+
+          ciphers = mkOption {
+            type = types.nullOr (types.listOf types.str);
+            default = null;
+            description = ''
+              Restrict the allowed ciphers of this policy to those defined
+              here. The default ciphers are those of the JVM platform.
+            '';
+          };
+
+          clientAuth = mkOption {
+            type = types.enum [ "NONE" "OPTIONAL" "REQUIRE" ];
+            default = "REQUIRE";
+            description = ''
+              The client authentication stance for this policy.
+            '';
+          };
+
+          privateKey = mkOption {
+            type = types.str;
+            default = "private.key";
+            description = ''
+              The name of private PKCS #8 key file for this policy to be found
+              in the <literal>baseDirectory</literal>, or the absolute path to
+              the key file. It is mandatory that a key can be found or generated.
+            '';
+          };
+
+          publicCertificate = mkOption {
+            type = types.str;
+            default = "public.crt";
+            description = ''
+              The name of public X.509 certificate (chain) file in PEM format
+              for this policy to be found in the <literal>baseDirectory</literal>,
+              or the absolute path to the certificate file. It is mandatory
+              that a certificate can be found or generated.
+
```

Comment:
To get the docs rendered as intended, these spaces need to have a

```
</para>
<para>
```

Also applies to the other options where this is used.